src/ImageService.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * HTTP-service for converting first page of a PDF document to a JPG image
+ * using Ghostscript.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2019-2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  PDF
+ * @author   Samuli Sillanpää <samuli.sillanpaa@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/configuration:external_content Wiki
+ */
+class ImageService
+{
+    /**
+     * Base directory for downloaded images and host status files.
+     *
+     * @var string
+     */
+    protected $baseDir;
+
+    /**
+     * Service for handling host blocking.
+     *
+     * @var \HostStatus
+     */
+    protected $hostStatus;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->baseDir = sys_get_temp_dir();
```

Comment:
Olsiko hyvä laittaa johonkin alihakemistoon, ettei cron:ssa tartte tyhjentää koko tmp:tä?

Reply:
Joo, parempi näin. Erottelin samalla alkuperäiset ja konvertoidut kuvat sekä status-tiedostot omiin alihakemistoihinsa /tmp/pdf2jpg:n alle.